### PR TITLE
Add ztest for kernel event logger.

### PR DESCRIPTION
**Pull request by SavinayDharmappa**

### CODE REVIEW

File: subsys/logging/kernel_event_logger.c [hunk 1]
```diff
@@ -61,7 +61,7 @@ SYS_INIT(_sys_k_event_logger_init,
  * to point to an application-defined routine.
  *
  */
-sys_k_timer_func_t _sys_k_get_time = k_cycle_get_32;
+sys_k_timer_func_t _sys_k_get_time = _timer_cycle_get_32;
```
Comment (lpereira) [inline subsys/logging/kernel_event_logger.c]: What is the build error this is fixing?

**Review by lpereira — COMMENTED** [review]
I'm having a hard time figuring out what this is testing exactly.  What are you looking at when things fail?  What are you looking at when things are working as they should?

File: tests/subsys/logging/kernel_event_logger/src/main.c [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			K_THREAD_STACK_SIZEOF(threadB_stack),
+			(k_thread_entry_t)threadB,
+			NULL, NULL, NULL, K_PRIO_PREEMPT(2), 0, 0);
+
+	k_thread_create(&threadA_data, threadA_stack,
+			K_THREAD_STACK_SIZEOF(threadA_stack),
+			(k_thread_entry_t)threadA,
+			NULL, NULL, NULL, K_PRIO_PREEMPT(2), 0, 0);
+
+	event_logger();
```
Comment (nashif) [inline tests/subsys/logging/kernel_event_logger/src/main.c]: shouldnt this be started is coop thread?

File: tests/subsys/logging/kernel_event_logger/testcase.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+tests:
+  test:
+    platform_whitelist: qemu_x86
```
Comment (nashif) [inline tests/subsys/logging/kernel_event_logger/testcase.yaml]: why are we whitelisting only qemu?

File: tests/subsys/logging/kernel_event_logger/src/main.c [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		case KERNEL_EVENT_LOGGER_THREAD_EVENT_ID:
+			printk("thread = %x, is moved to = %s ,at time = %d\n"
+			, event_data[1], event_type[event_data[2]], timestamp);
+			break;
+		default:
+			event_id = UNKNOWN_EVENT;
+			break;
+		}
+
+		zassert_not_equal(UNKNOWN_EVENT, event_id, "Unknown event");
```
Comment (nashif) [inline tests/subsys/logging/kernel_event_logger/src/main.c]: that is the only test I see in here, how are we verifying all events you have below?

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
closing, this interface will be deprecated.

**Pull request closed without merging**